controllers/activemqartemis_reconciler.go

```diff
@@ -1554,14 +1566,43 @@ func addNewVolumeMounts(existingNames map[string]string, existing *[]corev1.Volu
 	}
 }
 
-func (r *ActiveMQArtemisReconcilerImpl) MakeVolumeMounts(customResource *brokerv1beta1.ActiveMQArtemis, namer common.Namers) []corev1.VolumeMount {
+func (r *ActiveMQArtemisReconcilerImpl) MakeVolumeMounts(customResource *brokerv1beta1.ActiveMQArtemis, namer common.Namers) ([]corev1.VolumeMount, error) {
 
 	volumeMounts := []corev1.VolumeMount{}
 	if customResource.Spec.DeploymentPlan.PersistenceEnabled {
 		persistentCRVlMnt := volumes.MakePersistentVolumeMount(customResource.Name, namer.GLOBAL_DATA_PATH)
 		volumeMounts = append(volumeMounts, persistentCRVlMnt...)
 	}
 
+	for _, volume := range customResource.Spec.DeploymentPlan.ExtraVolumes {
+		var volumeMount *corev1.VolumeMount = nil
+		for _, vm := range customResource.Spec.DeploymentPlan.ExtraVolumeMounts {
+			if vm.Name == volume.Name {
+				volumeMount = vm
+				if volumeMount.MountPath == "" {
+					volumeMount.MountPath = volumes.GetDefaultMountPath(&volume)
```

Comment:
the name of a volume must be a DNS_LABEL and unique within the pod, see [Volume v1 core](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#volume-v1-core) so the base dire of the default mount path for extra volumes could be the same for all volume types, i.e. for an extra volume with the name `mydata`, the default path could be:
```
/amq/extra/volumes/mydata
```

In this way the implementation will not require updates to handle new volume types.

Reply:
I was thinking to give the default mount path some information of the volume types that may be useful for users. However that's not necessary as user have the option to specify the path of their own choices. Having said that I do agree that use a common base path is much simpler! will update. Thanks.